### PR TITLE
APIMF 3437

### PR DESCRIPTION
**Pull request by tomsfernandez**
Related to: https://github.com/aml-org/amf-aml/pull/405

- Simplified dialect lookup by id in CLI
- Chore: Removed TODOs that were not relevant or applicable or that ended up with a ticket created for them. Removed test file that was all commented out.

### CODE REVIEW

File: amf-api-contract/shared/src/main/scala/amf/apicontract/client/platform/AMFElementClient.scala [hunk 1]
```diff
@@ -32,12 +32,10 @@ class AMFElementClient private[amf] (private val _internal: InternalAMFElementCl
   override def renderToBuilder[T](element: DomainElement, builder: DocBuilder[T]): Unit =
     AMFElementRenderer.renderToBuilder(element, builder, getConfiguration())
 
-  // TODO ARM: Replace profile for spec
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/client/platform/AMFElementClient.scala:35]: Created ticket

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/plugins/ApiContractFallbackPlugin.scala [hunk 2]
```diff
@@ -16,8 +16,6 @@ import amf.core.internal.remote.{JSONRefs, Spec}
 import amf.core.internal.utils.MediaTypeMatcher
 import amf.core.internal.validation.CoreParserValidations.{CantReferenceSpecInFileTree, CouldntGuessRoot}
 
-// TODO ARM: change this, refactor to have different options based on the configuration (Raml strict, WebApi Relaxed).
-// TODO ARM: Export withFallback and this object to the user. User cannot create instance or implement interface? yes? no?
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/plugins/ApiContractFallbackPlugin.scala:20]: Created ticket

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/plugins/ApiContractFallbackPlugin.scala [hunk 3]
```diff
@@ -16,8 +16,6 @@ import amf.core.internal.remote.{JSONRefs, Spec}
 import amf.core.internal.utils.MediaTypeMatcher
 import amf.core.internal.validation.CoreParserValidations.{CantReferenceSpecInFileTree, CouldntGuessRoot}
 
-// TODO ARM: change this, refactor to have different options based on the configuration (Raml strict, WebApi Relaxed).
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/plugins/ApiContractFallbackPlugin.scala:19]: Obsolete, class already does this.

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/oas/OasParsePlugin.scala [hunk 4]
```diff
@@ -30,6 +30,5 @@ trait OasParsePlugin extends OasLikeParsePlugin {
                         wrapped: ParserContext,
                         ds: Option[OasWebApiDeclarations] = None): OasWebApiContext
 
-  // TODO ARM: Decide, should an OAS Document be able to reference another OAS Document?
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/oas/OasParsePlugin.scala:33]: Created ticket to clarify this with the expected behaviour.

File: amf-cli/shared/src/test/scala/amf/compiler/AMFCompilerTest.scala [hunk 5]
```diff
@@ -57,7 +57,7 @@ class AMFCompilerTest extends AsyncFunSuite with CompilerTestBuilder {
         s"file://amf-cli/shared/src/test/resources/reference-itself.raml",
         Raml10YamlHint,
         defaultConfig
-          .withErrorHandlerProvider(() => UnhandledErrorHandler), // TODO ARM then default should not throw exception?
```
Comment (tomsfernandez) [inline amf-cli/shared/src/test/scala/amf/compiler/AMFCompilerTest.scala:60]: Is not the default, obsolete TODO.

File: amf-cli/shared/src/test/scala/amf/validation/FromJsonLDPayloadValidationTest.scala [hunk 6]
```diff
@@ -1,87 +0,0 @@
-// TODO: ARM uncomment
-//package amf.validation
```
Comment (tomsfernandez) [inline amf-cli/shared/src/test/scala/amf/validation/FromJsonLDPayloadValidationTest.scala:2]: Obsolete, removed.

File: amf-shapes/shared/src/main/scala/amf/shapes/client/scala/ShapesConfiguration.scala [hunk 7]
```diff
@@ -185,7 +185,6 @@ object ShapesConfiguration {
 
   def predefined(): ShapesConfiguration = {
     ShapesRegister.register() // TODO ARM remove when APIMF-3000 is done
-    // TODO ARM: validate plugin and payload plugin of api?
```
Comment (tomsfernandez) [inline amf-shapes/shared/src/main/scala/amf/shapes/client/scala/ShapesConfiguration.scala:188]: Obsolete, payload plugin is already registered.

File: amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/emitter/JsonSchemaSerializer.scala [hunk 8]
```diff
@@ -10,7 +10,6 @@ trait JsonSchemaSerializer extends PlatformSecrets {
   // todo, check if its resolved?
   // todo lexical ordering?
 
-  // TODO ARM: change name, not descriptive.
```
Comment (tomsfernandez) [inline amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/emitter/JsonSchemaSerializer.scala:13]: Couldn't think of a better options. Besides I would have had to modify a lot of things. Isn't worth it.

File: amf-api-contract/shared/src/main/scala/amf/apicontract/client/platform/AMFElementClient.scala (hunk 1 above)
Comment (AgustinBettati) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/client/platform/AMFElementClient.scala:35]: Would leave this or raise a ticket, profile name should not be used here.

**Review by AgustinBettati — APPROVED** [review]

**Review by nschejtman — APPROVED** [review]

File: amf-api-contract/shared/src/main/scala/amf/apicontract/client/platform/AMFElementClient.scala (hunk 1 above)
Comment (tomsfernandez) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/client/platform/AMFElementClient.scala:35]: yup, created ticket for this one, forgot to click "submit" button in my review.

**Pull request merged**